bot/exts/error_handler.py

```diff
@@ -0,0 +1,59 @@
+from discord import Colour, Embed
+from discord.ext.commands import (BadArgument, Cog, CommandError,
+                                  CommandNotFound, Context)
+
+from bot.bot import SirRobin
+from bot.log import get_logger
+
+log = get_logger(__name__)
+
+
+class ErrorHandler(Cog):
+    """Handles errors emitted from commands."""
+
+    def __init__(self, bot: SirRobin):
+        self.bot = bot
+
+    @staticmethod
+    def _get_error_embed(title: str, body: str) -> Embed:
+        """Return a embed with our error colour assigned."""
+        return Embed(
+            title=title,
+            colour=Colour.brand_red(),
+            description=body
+        )
+
+    @Cog.listener()
+    async def on_command_error(self, ctx: Context, error: CommandError) -> None:
+        """
+        Generic command error handling from other cogs.
+
+        Using the error type, handle the error appropriately.
+            if there is no handling for the error type raised,
+            a message will be sent to the user & it will be logged.
+
+        In the future, I would expect this to be used as a place
+            to push errors to a sentry instance.
+        """
+        log.trace(f"Handling a raised error {error} from {ctx.command}")
+
+        if isinstance(error, BadArgument):
+            embed = self._get_error_embed("Bad argument", str(error))
+            await ctx.send(embed=embed)
+            return
+        elif isinstance(error, CommandNotFound):
+            embed = self._get_error_embed("Command not found", str(error))
+            await ctx.send(embed=embed)
+            return
+
+        # If we haven't handled it by this point, it is considered an unexpected/handled error.
+        await ctx.send(
+            f"Sorry, an unexpected error occurred. Please let us know!\n\n"
+            f"```{error.__class__.__name__}: {error}```"
```

Comment:
I don't think we need this. It introduces a few problems. Let's just stick with the basic message, and keep the actual error in logs.

Reply:
What are the problems you are referring to?

I think because it runs on a server where we wouldn't beable to see the logs easily unless you have direct access, could cause unhandled errors to be harder to spot. This gives us a way to see them, so we can create issues and add handlers.

The codebase has nothing to hide from its users. I even took this snippet from the main bot [here](https://github.com/python-discord/bot/blob/main/bot/exts/backend/error_handler.py#L303-L306). I would expect the reasoning for this to be the same as mine.

Reply:
Fair point, we don't even have Sentry set up just now.